capstone/src/test/java/com/google/sps/CachedInterestVectorTest.java

```diff
@@ -145,4 +145,47 @@ public void renormalizeVectorHashMapTest() {
         Assert.assertEquals(vectorHashMap.get(Tag.COOKING.getTag()), (2.0/magnitude), epsilon);
         Assert.assertEquals(vectorHashMap.get(Tag.FITNESS.getTag()), (1.0/magnitude), epsilon);
     }
-}
+
+    @Test
+    public void addRoomUpdateCachedInterestVectorTest() {
+        User user = new User("123" , "John");
+        Room room = new Room(user, "Piano", "Piano Lessons");
+        room.addTag(Tag.EDUCATION);
+        room.addTag(Tag.MUSIC);
+        room.addTag(Tag.ART);
+
+        CachedInterestVector.addRoomUpdateCachedInterestVector(user, room);
+
+        EmbeddedEntity embeddedVectorMap = user.getCachedInterestVector();
+        Double expectedMagnitude = Math.sqrt(3.0);
+        Assert.assertEquals(user.getMagnitude(), expectedMagnitude, epsilon);
+        Assert.assertEquals((Double) embeddedVectorMap.getProperty(Tag.EDUCATION.getTag()), (1.0/expectedMagnitude), epsilon);
+        Assert.assertEquals((Double) embeddedVectorMap.getProperty(Tag.MUSIC.getTag()), (1.0/expectedMagnitude), epsilon);
+        Assert.assertEquals((Double) embeddedVectorMap.getProperty(Tag.ART.getTag()), (1.0/expectedMagnitude), epsilon);
+    }
+    
+    @Test
+    public void removeRoomUpdateCachedInterestVectorTest() {
+        User user = new User("123" , "John");
+        Room room = new Room(user, "Piano", "Piano Lessons");
+        room.addTag(Tag.EDUCATION);
+        room.addTag(Tag.MUSIC);
+        room.addTag(Tag.ART);
+        EmbeddedEntity embeddedVectorMap = new EmbeddedEntity();
+        Double magnitude = Math.sqrt(15.0);
```

Comment:
3^2 + 2^2 + 1^2 = 14. If this passes, perhaps your epsilon is a little to large? To avoid this kind of thing happening, I would re-use the same setup you use in your actual code, since each test can assume the other methods are propertly tested. So I would do something like:
```java
HashMap<String, Double> preexisting_tags = new HashMap<>();
preexisting_tags.put(...);
user.setCachedInterestVector(hashMapToEmbeddedEntity(preexisting_tags));
user.setMagnitude(magnitude(preexisting_tags));
```

Reply:
It passes since I am passing in the magnitude too (which you commented on another PR that the magnitude shouldn't be passed in) so it should be 14.0. Got it.